### PR TITLE
Update function following spikeinterface upgrade

### PR DESCRIPTION
**Pull request by ZhixiaoSu**
1. Updated package to take care of version difference between new and old versions of spikeinteraface and firmware. 

- 'openephys' changed to 'openephysbinary' in get_neo_streams

- spikeinterface version changed to 0.101.2 in pyproject

2. Misc

- Changed saving processing log to inside of function generate_qc_report, so that calling the function itself (instead of the whole script) will also save processing log. 
- Updated driftmap.json so that use can decide if to perform phaseshift or not. (Very time consuming.)

### CODE REVIEW

File: src/aind_ephys_rig_qc/qc_figures.py [hunk 1]
```diff
@@ -190,6 +190,7 @@ def plot_drift(directory, stream_name, block_index=0):
         "vmax": 0,
         "cmap": "Greys_r",
         "figsize": [10, 10],
+        "phase_shift": {"margin_ms": 100.0},
```
Comment (alejoe91) [inline src/aind_ephys_rig_qc/qc_figures.py]: maybe you can make `phase_shift` optional:
- if `"phase_shift"` is `None`, don't run it
- else, run it with params

File: src/aind_ephys_rig_qc/test.ipynb [hunk 2]
```diff
@@ -0,0 +1,83 @@
+{
+ "cells": [
```
Comment (alejoe91) [inline src/aind_ephys_rig_qc/test.ipynb]: was this file added on purpose?

File: src/aind_ephys_rig_qc/qc_figures.py (hunk 1 above)
Comment (ZhixiaoSu) [reply src/aind_ephys_rig_qc/qc_figures.py]: Fixed in the new version by adding an editable .json file.

File: src/aind_ephys_rig_qc/test.ipynb (hunk 2 above)
Comment (ZhixiaoSu) [reply src/aind_ephys_rig_qc/test.ipynb]: I removed this test file. It was added by accident.

**Comment by jsiegle** [comment]
This looks good, merging into `main`!

**Pull request merged**